### PR TITLE
docs: write project CLAUDE.md with layout, commands, and contribution conventions

### PR DESCRIPTION
**Pull request by castastrophe**
## What

Replace the effectively empty CLAUDE.md (a stray Cursor-style frontmatter block with no content) with real project instructions for coding agents.

## Why

Agents working in this repo had no guidance to load — the file carried zero information, so every session re-derived the project layout, commands, and conventions from scratch (or missed them entirely, e.g. the Weaver-managed README blocks).

## How

The new file is ~60 lines, organized so a smaller model can act on it without exploring first:

- **Orientation** — what the package is, plus the local-first / no-network-calls constraint stated as a rule to preserve
- **Layout** — the four flat root files (`index.js`, `cli.js`, `mcp.js`, `test.js`); explicit note that there is no `src/`
- **Commands** — `yarn test` / `coverage` / `typecheck` / `build:types`, with the Yarn 4 + Node ≥ 24 requirement
- **Conventions** — JSDoc-not-TypeScript (with the why: published types are generated from JSDoc), self-documenting code with brief why-only comments, the real-tempdir/real-git test pattern, dependency-light pitch, Weaver `weaver:*:START/END` markers, and the husky hook behavior
- **Commits and PRs** — Conventional Commits shape, the semantic-release type→bump mapping and body-becomes-release-notes contract, squash-merge PR title rule, and no AI-attribution footers

Everything documented was verified against the repo (package.json, configs, hooks, source) rather than written from assumption.

## Test plan

Docs-only change; no runtime surface. Verified the commit passed the pre-commit (lint-staged/prettier) and commit-msg (commitlint) hooks locally.

---

- [ ] Tests added or updated (`yarn test`) — n/a, docs only
- [x] Lint clean (`yarn eslint .`)
- [ ] Types pass (`yarn typecheck`) — n/a, docs only
- [x] Documentation updated where relevant
- [x] Conventional commit format used (enforced by commitlint)

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -1,4 +1,59 @@
...
+Conventional Commits, enforced by commitlint. semantic-release runs on `main`:
+the commit type drives the version bump (`feat` → minor, `fix` → patch,
+`BREAKING CHANGE:` footer → major) and the commit body is lifted verbatim into
+the release notes — write it for a human reader.
+
+- Subject: `<type>(<optional-scope>): <imperative subject>` — lowercase, ≤100
+  chars, no trailing period. Types: feat, fix, docs, chore, refactor, test, …
+- Body: plain prose, wrapped ~72–80 chars — what changed and why it matters;
+  for feature commits, also how it behaves and how edge cases are handled.
+  Small commits may collapse to subject + one line + footer (`Closes #N`).
```
Comment (castastrophe) [inline CLAUDE.md]:
```suggestion
- Body: plain prose, not wrapped, 1-2 sentences, <80 chars — what changed and why it matters;
  for feature commits, also how it behaves and how edge cases are handled.
  Small commits may collapse to subject + one line + footer (`Closes #N`).
```
